### PR TITLE
Improve close Popup behaviour

### PR DESCRIPTION
**Pull request by jouderianjr**
Adds possibility to close popup clicking outside of the popup or
using the escape key

💃

### CODE REVIEW

**Comment by rogeriochaves** [comment]
Awesome job! Thank you so much!

I've tested your changes, and noted a couple of things:
- We cannot put a simple `onClick` on the overlay, because it also closes when clicking inside the popup, since the popup is inside the overlay. We need to change the markdown structure so they are siblings or something like that
- The esc key works perfectly on facebook, but not on twitter, idk why, I've put a `Debug.log` and it seems that Elm triggers the esc key normally on twitter, except when the popup is open, very weird

**Comment by jouderianjr** [comment]
Thanks @rogeriochaves. You are completely right about the onClick thing, I'll refactor.

I didn't tested on twitter ( to be honest I thought that was only for facebook! nice to know 😄 ), I'll try to discover why is not working

**Comment by rogeriochaves** [comment]
@jouderianjr if you can't discover the twitter problem, it is fine, it wasn't there before anyway. Only the `onClick` thing is a problem, because it breaks the popup functionality :)

**Comment by jouderianjr** [comment]
I Just updated, could you test it again? Unfortunately, I didn't discovered why esc is not working on twitter :/

File: src/FlagPopup.elm [hunk 1]
```diff
@@ -126,10 +138,22 @@ popup model =
...
-            screen <|
-                el Overlay
-                    [ width (percent 100), height (percent 100) ]
-                    (modal NoStyle [ center, verticalCenter ] (modalContents model))
+            row NoStyle
+                []
+                [ el Overlay
+                    [ width (percent 100)
+                    , minHeight (percent 100)
+                    , inlineStyle [ ( "position", "fixed" ), ( "z-index", "999" ) ]
```
Comment (rogeriochaves) [inline src/FlagPopup.elm:146]: nice, didn't knew about this, is it the same that `screen` does or not?

Comment (rogeriochaves) [reply src/FlagPopup.elm:146]: oh I see, you need to handle the z-indexes manually

**Comment by rogeriochaves** [comment]
this is PERFECT! Thank you so much! Much better ux :D

(i kept trying to press ESC myself but was too lazy to implement this)

**Comment by rogeriochaves** [comment]
[Published](https://circleci.com/gh/fake-news-detector/extension/28) as v0.11, should be on the stores soon

**Pull request merged**